service/worker/workerdeployment/version_workflow.go

```diff
@@ -235,6 +238,18 @@ func (d *VersionWorkflowRunner) handleUpdateVersionMetadata(ctx workflow.Context
 }
 
 func (d *VersionWorkflowRunner) startDrainage(ctx workflow.Context, isCan bool) {
+	v := workflow.GetVersion(ctx, "Step1", workflow.DefaultVersion, 1)
+	if v != workflow.DefaultVersion { // needs patching because we added a Signal call via d.syncSummary
+		if d.VersionState.GetDrainageInfo().GetStatus() == enumspb.VERSION_DRAINAGE_STATUS_UNSPECIFIED {
+			now := timestamppb.New(workflow.Now(ctx))
+			d.VersionState.DrainageInfo = &deploymentpb.VersionDrainageInfo{
+				Status:          enumspb.VERSION_DRAINAGE_STATUS_DRAINING,
+				LastChangedTime: now,
+				LastCheckedTime: now,
+			}
+			d.syncSummary(ctx)
```

Comment:
should we set the drainage status in the Deployment wf too (in VersionSummary) when a version stops being ramping/current? In case some one is using that as checking drainage status. Does not have to go to this PR.

Reply:
done